src/crimson/osd/main.cc

```diff
@@ -85,6 +116,10 @@ int main(int argc, char* argv[])
                                               CEPH_ENTITY_TYPE_OSD,
                                               &cluster_name,
                                               &conf_file_list);
+
+  Preforker forker;
```

Comment:
why do you prefer `Preforker` over `daemon(3)` call?

Reply:
do you know how to use daemon(3) to implement --foreground?

Reply:
please refer ceph_mgr.cc .

Reply:
sure, will update the patch, BTW, do we need update chdir and pid_file? if so, it is set in config file or through command line options?  at the very beginning, we didn't parse config file.

Reply:
@tchaikov updated, please review.